tests/integration/test_bayesian_optimization.py

```diff
@@ -232,46 +300,103 @@ def _test_optimizer_finds_minimum(
 ) -> None:
 
     model_args = model_args or {}
-    search_space = BRANIN_SEARCH_SPACE
-
-    def build_model(data: Dataset) -> GPflowPredictor:
-        assert model_args is not None
-
-        variance = tf.math.reduce_variance(data.observations)
-        kernel = gpflow.kernels.Matern52(variance, tf.constant([0.2, 0.2], tf.float64))
-        scale = tf.constant(1.0, dtype=tf.float64)
-        kernel.variance.prior = tfp.distributions.LogNormal(
-            tf.constant(-2.0, dtype=tf.float64), scale
-        )
-        kernel.lengthscales.prior = tfp.distributions.LogNormal(
-            tf.math.log(kernel.lengthscales), scale
-        )
-
-        if model_type == "GPR":
-            gpr = gpflow.models.GPR(
-                (data.query_points, data.observations), kernel, noise_variance=1e-5
+    num_initial_query_points = 5
+    track_state = True
+
+    if optimize_branin:
+        search_space = BRANIN_SEARCH_SPACE
+        minimizers = BRANIN_MINIMIZERS
+        minima = SCALED_BRANIN_MINIMUM
+        rtol_level = 0.005
+    else:
+        search_space = SIMPLE_QUADRATIC_SEARCH_SPACE
+        minimizers = SIMPLE_QUADRATIC_MINIMIZER
+        minima = SIMPLE_QUADRATIC_MINIMUM
+        rtol_level = 0.05
+
+    if model_type in ["GPR", "VGP", "SVGP"]:
+
+        def build_model(data: Dataset) -> GPflowPredictor:  # type: ignore
```

Comment:
Do talk to me if you hit typing issues after you merge in my recent PR.

Reply:
all seems good after your merge - I didn't introduce those new build functions now, better left for a separate PR and doing it properly